x-pack/plugins/security/server/authorization/privileges/privileges.ts

```diff
@@ -97,10 +133,49 @@ export function privilegesFactory(
           }
         }
 
-        if (Object.keys(featurePrivileges[feature.id]).length === 0) {
-          delete featurePrivileges[feature.id];
+        if (feature.disabled || Object.keys(featurePrivileges[feature.id]).length === 0) {
+          disabledFeatures.add(feature.id);
         }
       }
+
+      // Update composable feature privileges to include and deduplicate actions from the referenced privileges.
+      // Note that we should do it _before_ removing disabled features.
+      for (const composableFeature of composableFeaturePrivileges) {
```

Comment:
This doesn't support our licensing system, which makes it possible to compose a feature using privileges that shouldn't be available at the current license level. I think this is OK for our immediate needs, but perhaps it's worth adding a comment calling this out. What do you think?

Reply:
That's a very good point! Yes, will add a comment for now, but later we might need to implement a validation if we don't want to support that.